File: JumpGame.java
// https://leetcode.com/problems/jump-game/
class JumpGame{
    public boolean canJump(int[] nums) {
        int len=nums.length;
        int d = len-1;
        for(int i=len-2;i>=0;i--){
            if(nums[i]+i>=d)
                d=i;
        }
        return d==0;
    }
}

File: JumpGame.py
# https://leetcode.com/problems/jump-game
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        goal=len(nums)-1
        for i in range(len(nums)-1,-1,-1):
            if(i+nums[i]>=goal):
                goal=i #shift goal to i'th position
        return goal==0 #two possibilities at the end. goal=0 meaning the goal can be reached or goal!=0 meaning the goal cant be reached.